scene-recognition/process.py

```diff
@@ -38,46 +53,99 @@ def format_timecode(value):
-        # output to csv with same filename
-        df.to_csv(destination, index=False)
+            'modifier': 'transitional'})
+    # remove filename, transcript and note columns
+    df = df.drop('filename', axis=1)
+    df = df.drop('transcript', axis=1)
+    df = df.drop('note', axis=1)
+    return df
 
 
-if __name__ == '__main__':
-    task_dir = pathlib.Path(__file__).parent
-    golds_dir = task_dir / 'golds'
+def copy_timeframes(source, destination):
+    """This was used at some point to create timeframes from timepoints, but we are not
+    storing derivable data anymore. Keeping it around for a while in case it is useful,
+    but this function will be removed at some point."""
+    df = pd.read_csv(source)
+    df = amend_dataframe(df)
+    current_label = None
+    current_sublabel = None
+    current_frame = []
+    with open(destination, "w") as csv_file:
+        csv_writer = csv.writer(csv_file)
+        csv_writer.writerow(_get_column_names())
+        for index, row in df.iterrows():
+            ts = row['timestamp']
+            label = row['type label']
+            sublabel = row['subtype label']
+            sublabel = '-' if type(sublabel) is float else sublabel
+            modifier = row['modifier']
+            if _labels_match(current_label, label, current_sublabel, sublabel):
+                current_frame.append((ts, label, sublabel))
+            else:
+                if current_frame and current_frame[0][1] != '-':
+                    # this takes the modifier of the last timepoint
+                    timeframe = _create_timeframe(
+                        current_frame, current_label, current_sublabel, modifier)
+                    csv_writer.writerow(timeframe)
+                current_frame = [(ts, label, sublabel)]
+                current_label = label
+                current_sublabel = sublabel
+
 
-    # delete golds directory if it exists
-    shutil.rmtree(golds_dir, ignore_errors=True)
-    # then start from clean slate
-    golds_dir.mkdir(exist_ok=True)
+def _get_column_names():
+    """Helper method for copy_timeframes."""
+    if USE_SUBLABEL:
+        return ['start', 'end', 'type label', 'subtype label', 'modifier']
+    return ['start', 'end', 'type label', 'modifier']
+
+
+def _labels_match(current_label, label, current_sublabel, sublabel):
+    """Helper method for copy_timeframes."""
+    if USE_SUBLABEL:
+        return current_label == label and current_sublabel == sublabel
+    else:
+        return current_label == label
+
+
+def _create_timeframe(frame, label, sublabel, modifier):
+    """Helper method for copy_timeframes."""
+    sub = [sublabel] if USE_SUBLABEL else []
+    return [frame[0][0], frame[-1][0], label] + sub + [modifier]
+
+
+
+if __name__ == '__main__':
 
-    # find all directories starts with six digits and a dash
-    for batch_dir in task_dir.glob('[0-9][0-9][0-9][0-9][0-9][0-9]-*'):
-        process(batch_dir, golds_dir)
+    folder = Path.cwd()
```

Comment:
I guess gold files must be placed in the same directory where this script is located, instead of wherever user decided to run the script.

Reply:
Yes, that 's true. I did not change that party of the code and I will leave it like that for now but change it if we have a general requirements for process files that. Or you can go ahead an do it.